### PR TITLE
Added ScanIntoItems and ScanIntoItem functions

### PR DESCRIPTION
**Pull request by moltzaum**
## Which issue is fixed by this PR? If not related to an existing issue, what does this PR do?

First I created a ScanIntoItems function that basically unpacks a list
into sql's scan function.

Then I updated the generic cruder to use the function, although it is meant
more for endpoints that don't already have a generic function. I updated
the generic case since it could be used as a model and so that the error
messages won't look too different from implementors that use ScanIntoItems.

ScanIntoItems is especially convienent when returning a number of items.
It could potentially be used for: #2494

Here is an example that is applicable for cachegroups:
```
items := []interface{}{
    &cg.ID,
    &cg.ParentName,
    &cg.SecondaryParentName,
}
if err = api.ScanIntoItems(resultRows, items); err != nil {
    if err == sql.ErrNoRows {
	return errors.New("no cachegroup found with this id"), nil, http.StatusNotFound
    }
    return nil, err, http.StatusInternalServerError
}
```

Mayber later down the line, ScanIntoItems would be prefered over
ScanStruct since ScanStruct belongs to sqlx. I don't think we use sqlx
for anything other than ScanStruct, so we could potentially get rid of
a dependency.

## Which TC components are affected by this PR?

- Traffic Ops

### (If no tests are part of this PR, please provide explanation as to why no tests are included.)

Run the traffic ops integration tests.

This PR doesn't introduce new tests, but probably should. The function advises function callers that they don't need to close sql.Rows. The ability of the function to close the rows properly should be tested, but I'm not sure how it should be done yet. I found the mockdb, but I haven't found an example of a query running against it yet.

## Check all that apply

- [ ] This PR includes tests
- [x] This PR includes documentation updates
- [ ] This PR includes an update to CHANGELOG.md
- [x] This PR includes all required license headers
- [x] This PR does not include a database migration
- [x] This PR does not fix a serious security flaw. Read more: [www.apache.org/security](http://www.apache.org/security/)

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/3398/
Test PASSed.

**Comment by moltzaum** [comment]
Update:
As I've written it now, `ScanIntoItems` uses `sql.Rows`, but an update or create should use `QueryRow`, instead of `NamedQuery` or `Query`.

The generic create uses `NamedQuery` only because it allows us to map fields from an interface using their names. Note: is possible to have a simple `NamedQueryRow` function (https://github.com/jmoiron/sqlx/issues/102), or to make a function that uses reflection to read from the struct in a linear fashion (instead of using names). Both of these solutions would eliminate the need for `ScanIntoIItems`

TL;DR:
I don't think `ScanIntoItems` should be used as it currently exists, so I'm closing this PR.
